docs/architecture/decisions/0006-unique-ids-for-private-beta.md

```diff
@@ -41,24 +41,28 @@ Thus a valid LPA number would be M-0000-0000-0013 where the 3 is a checkdigit ge
 
 Typical tranposing issues such as digit order twiddling. i.e. 0000-0000-1234 is a correct checkdigit of 4, but 0000-0000-2134 would be invalid detecting the 12 to 21 swap.
 
-We have estimated scenarios for potential ID usage over a 10 year period at current and aggressive growth. (see [internal spreadsheet](https://docs.google.com/spreadsheets/d/1kH1BOk8iOTUg3aEphD5J18LV4zhr4mQn/edit?usp=sharing&ouid=104236485660673640558&rtpof=true&sd=true)).
+We have estimated scenarios for potential ID usage over a 10 year period at current and aggressive growth. (see [spreadsheet attached to issue MLPA-2450](https://opgtransform.atlassian.net/browse/MLPAB-2450)).
 
 ## Consequences
 
 - Reducing the information space around the reference number to only the 0-9 characters with a check digit means we have a maximum of 99,999,999,999 numbers available.
 
-- We will generate references non-sequentially, so that it is easier to detect enumeration attacks on any service that uses the ref number as an ID. i.e. 0001 is not followeed by 0002 and our approach should inject sufficient padding to make iteration attacks easily detectable.
+- We will generate references non-sequentially, so that it is easier to detect enumeration attacks on any service that uses the ref number as an ID. i.e. 0001 is not followeed by 0002 and our approach should inject sufficient padding to make iteration attacks easily detectable and avoid predictable step jumps.
 
-- LPA reference numbers that are purely numeric are more guessable with enough tries or given sufficinet requests. They should not be used as publically facing identifiers in web based services without some 2nd factor (for example the paper verification codes) and any services using them as url identifiers or form parameters should put in place detection for brute forcing and iteration based attacks.
+- LPA reference numbers that are purely numeric are more guessable with enough tries or given sufficient requests. They should not be used as publicly facing identifiers in web based services without some 2nd factor (for example the paper verification codes) and any services using them as url identifiers or form parameters should put in place detection for brute forcing and iteration based attacks.
 
-- They should not be used in URLs on external-facing systems.
+- They should not be used in URLs on external-facing systems. In external facing services or APIs that need to make direct reference to an LPA the UUID associated with an LPA should be used instead.
```

Comment:
LPAs don't currently have a UUID. There's a UUID created in MRLPA before the LPA is assigned a reference number, but it's not shared into any other systems.

At the moment, are "external facing services or APIs" only MRLPA and Use?

Reply:
In terms of External-facing I was thinking more the MrLPA API that will go live for supporting orgs at some point, rather than internal OPG systems like MrLPA and Use.